crates/red_knot_project/src/db/changes.rs

```diff
@@ -201,9 +208,16 @@ impl ProjectDatabase {
                         return WalkState::Continue;
                     }
 
-                    let mut paths = added_paths.lock().unwrap();
+                    if entry
+                        .path()
+                        .extension()
+                        .and_then(PySourceType::try_from_extension)
+                        .is_some()
+                    {
+                        let mut paths = added_paths.lock().unwrap();
 
-                    paths.push(entry.into_path());
+                        paths.push(entry.into_path());
+                    }
```

Comment:
I got really confused when red knot started parsing my `knot.toml` file as a python file. For now, I did the simplest fix by copying this condition from `Project::discover_files`. I have to rework this code next month anyway to properly support file inclusion and exclusions.